schedule/scheduler/scheduler.go

```diff
@@ -55,11 +59,24 @@ func (s *Scheduler) Schedule(start, stop time.Time) (*schedule.Schedule, error)
 
 // ExtendSchedule takes a previously generated schedule and extends it. The user rotation continues as normal from the
 // last shift in the schedule.
-func (s *Scheduler) ExtendSchedule(sched *schedule.Schedule, stopInclusive time.Time) error {
+//
+// If prune is true:
+// * all shifts completed in the past are purged, and all current and future shifts are checked for owners in the
+// current rotation.
+// * If a shift is assigned to a user that is no longer in the rotation (including if the missing user is in the
+// userOverride field), that shift and all future shifts are rescheduled.
+// * Rescheduled shifts do not carry over previous userOverride values.
```

Comment:
Hopefully folks won't be added/removed from the rotation too frequently, but I guess we should make sure to communicate every time this does happen that any previously rescheduled shifts are lost...otherwise my assumption would have been that as a rotation member who got my vacation days covered, I could just forget about it.

Reply:
Agreed - I didn't know of a good way to handle this, but I figured since the action using this should open a PR before being submitted, a user removal change may be a little more obvious. We'll find out soon, I guess.

Reply:
Along similar lines, how often will we be clearing the past schedule?  I guess it's not the most important feature, but it is sometimes nice to keep at least some history in the schedule...if only so that when I feel like I was just on rotation 2 days ago I can look back and realize it was longer.

Reply:
I imagine extending & pruning the schedule probably on the 1st and 15th of the month (approx every two weeks). Once the rotation is extended from 1 day to (likely) 7 days, you probably won't have that "I was just on 2 days ago" feeling.